### PR TITLE
Create debian package

### PR DESCRIPTION
**Pull request by ThijsWithaar**
Generate a debian package, with correct folder names.
Also fixed a compiler warning and linker error on debian

### CODE REVIEW

**Comment by bragadeesh** [comment]
The travis CI build is failing like before due to DEB, can you see why?
https://travis-ci.org/clMathLibraries/clFFT/jobs/307157987

**Comment by ThijsWithaar** [comment]
Mmm, the key line seems to be:
"CPack Error: Cannot initialize CPack generator: DEB"

I assumed that the .deb generator would be available on other linux systems as well, this might not be the case. I'll go google a bit.

**Review by bragadeesh — COMMENTED** [review]
What is the need for adding empty destructor for FFTAction?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -107,11 +107,11 @@ set( SUFFIX_LIB_DEFAULT "" )
 set( SUFFIX_BIN_DEFAULT "" )
 
 # Modify the global find property to help us find libraries like Boost in the correct paths for 64-bit
-# Essentially, find_library calls will look for /lib64 instead of /lib; works for windows and linux
+# Essentially, find_library calls will look for /lib64 instead of /lib; works for windows, not on debian
 if( BUILD64 )
 	set_property( GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE )
 	message( STATUS "64bit build - FIND_LIBRARY_USE_LIB64_PATHS TRUE" )
-	if( NOT APPLE )
+	if( WIN32 )
```
Comment (bragadeesh) [inline src/CMakeLists.txt]: Non debian linux system like CentOS still use lib64 folder structure that we need to support

File: src/library/enqueue.cpp [hunk 2]
```diff
@@ -90,13 +90,18 @@ FFTStockhamAction::FFTStockhamAction(clfftPlanHandle plHandle, FFTPlan * plan, c
...
 
-
 FFTAction::FFTAction(FFTPlan * fftPlan, clfftStatus & err)
     : plan(fftPlan)
 {
     err = CLFFT_SUCCESS;
 }
 
+
+FFTAction::~FFTAction()
```
Comment (bragadeesh) [inline src/library/enqueue.cpp:100]: what is the need for empty destructor?

Comment (ThijsWithaar) [reply src/library/enqueue.cpp:100]: It's a base class, with virtual functions, so it needs the declaration so the destructor of the derived classes gets called. The implementation here is to make the linker happy.

File: src/CMakeLists.txt (hunk 1 above)
Comment (ThijsWithaar) [reply src/CMakeLists.txt]: Ah, so then it's better to explicitly check for debian systems, I suppose

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -337,12 +348,20 @@ endif( )
 
 set( CPACK_SOURCE_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}-${CLFFT_VERSION}-${CMAKE_HOST_SYSTEM_NAME}-Source" )
 
+# The following code is setting variables to control the behavior of CPack to generate our
+set( CPACK_PACKAGE_CONTACT "none@gmail.com")
```
Comment (bragadeesh) [inline src/CMakeLists.txt]: Is the 'CPACK_PACKAGE_CONTACT' with email required, can you remove this line?

Comment (ThijsWithaar) [reply src/CMakeLists.txt]: It is required for the packaging to succeed, it won't build without it.

Comment (bragadeesh) [reply src/CMakeLists.txt]: Looks like the email id is some random one that I like to avoid, can you leave it empty? like
set( CPACK_PACKAGE_CONTACT "")

Comment (ThijsWithaar) [reply src/CMakeLists.txt]: Leaving it with an empty string gives the following error:

```
CMake Error at /usr/share/cmake-3.9/Modules/CPackDeb.cmake:820 (message):
  CPackDeb: Debian package requires a maintainer for a package, set
  CPACK_PACKAGE_CONTACT or CPACK_DEBIAN_PACKAGE_MAINTAINER
Call Stack (most recent call first):
  /usr/share/cmake-3.9/Modules/CPackDeb.cmake:1049 (cpack_deb_prepare_package_vars)
```

Comment (bragadeesh) [reply src/CMakeLists.txt]: Does the string have to be an email? Can you try "clFFT maintainers"

Comment (ThijsWithaar) [reply src/CMakeLists.txt]: Indeed, your suggestion makes it work, see the commit below.

**Comment by bragadeesh** [comment]
thanks

**Pull request merged**